canonical-web-and-design/maas-ui
# feat(ui): update machine config form to latest design

## Done

- Simplified the code now that the form gets unmounted, so we don't have to manage the internal form values as much
- Removed tags from "Machine configuration" form
- Centered form fields
- Migrated `MachineForm.test.tsx` to RTL

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the configuration tab of a machine and check that it matches the [design]() for:
  - Large screen read-only
  - Large screen edit
  - Small screen read-only
  - Small screen edit

## Fixes

Fixes canonical-web-and-design/app-tribe#708
Fixes canonical-web-and-design/app-tribe#744

## Screenshots

### Read-only
![2022-03-30_17-28](https://user-images.githubusercontent.com/25733845/160776216-0a7c0844-e2ca-49d7-a46f-608b7fe9bf3c.png)

### Edit
![2022-03-30_17-29](https://user-images.githubusercontent.com/25733845/160776235-2bcc7fdf-418d-4bb1-9adb-15858d5328b6.png)
